Sources/AsyncHTTPClient/AsyncAwait/HTTPClient+execute.swift

```diff
@@ -112,46 +126,76 @@ extension HTTPClient {
                 return response
             }
 
+            resendCount += 1
             currentRequest = newRequest
         }
     }
 
     private func executeCancellable(
         _ request: HTTPClientRequest.Prepared,
         deadline: NIODeadline,
-        logger: Logger
+        logger: Logger,
+        context: ServiceContext,
+        resendCount: Int?
     ) async throws -> HTTPClientResponse {
         let cancelHandler = TransactionCancelHandler()
 
-        return try await withTaskCancellationHandler(
-            operation: { () async throws -> HTTPClientResponse in
-                let eventLoop = self.eventLoopGroup.any()
-                let deadlineTask = eventLoop.scheduleTask(deadline: deadline) {
-                    cancelHandler.cancel(reason: .deadlineExceeded)
-                }
-                defer {
-                    deadlineTask.cancel()
-                }
-                return try await withCheckedThrowingContinuation {
-                    (continuation: CheckedContinuation<HTTPClientResponse, Swift.Error>) -> Void in
-                    let transaction = Transaction(
-                        request: request,
-                        requestOptions: .fromClientConfiguration(self.configuration),
-                        logger: logger,
-                        connectionDeadline: .now() + (self.configuration.timeout.connectionCreationTimeout),
-                        preferredEventLoop: eventLoop,
-                        responseContinuation: continuation
-                    )
-
-                    cancelHandler.registerTransaction(transaction)
-
-                    self.poolManager.executeRequest(transaction)
-                }
-            },
-            onCancel: {
-                cancelHandler.cancel(reason: .taskCanceled)
+        return try await withSpan(request.head.method.rawValue, context: context, ofKind: .client) { span in
```

Comment:
Should the `Tracer` be a property of the client, to allow for easier unit testing? And just default it to the current process-bootstrapped tracer in the initializer, unless it's overridden?

Reply:
We could do that, yeah. Elsewhere we just use `InstrumentationSystem.bootstrapInternal` in unit tests which also works fine but I'd be happy either way.

Reply:
Unless there are downsides, I prefer the explicit dependency injection. Easy enough to debug and reason about.

Reply:
Sounds good to me as long as we don't expose this initializer publicly.

Reply:
What's the downside of adding it to the public initializer?

Reply:
It goes against the way Swift Distributed Tracing usually works, where a single tracer is chosen for the system and used transparently everywhere. Having said that, Logging also allows passing specific `Logger`s around and still use `LoggingSystem`. I'm curious to hear what @ktoso thinks about this.